rfc-087-dealing-with-errors.md

```diff
@@ -0,0 +1,105 @@
+# Dealing with errors
+
+## Summary
+
+This describes how we treat errors on GOV.UK
+
+## Problem
+
+We've recently migrated to a new error tracking service, Sentry. This provides us an opportunity to rethink how we treat errors.
+
+## Proposal
+
+There are 2 principles:
+
+### 1. When something goes wrong, we should be notified
+
+Applications should report exceptions to Sentry. Applications must not swallow errors.
+
+### 2. Applications should not error
+
+The goal of GOV.UK is that applications should not error. When something goes wrong it should be fixed.
+
+## Classifying errors
+
+### Bug
+
+A code change makes the application crash.
+
+Desired behaviour: error is sent to Sentry, developers are notified and fix the error. Developers mark the error in Sentry as `Resolved`. This means a recurrence of the error will alert developers again.
+
+### Intermittent errors without user impact
+
+Frontend applications often see timeouts when talking to the content-store.
+
+There's no or little user impact because the request will be answered by the caching layer.
+
+Example: <https://sentry.io/govuk/app-finder-frontend/issues/352985400>
+
+Desired behaviour: error is not sent to Sentry. Instead, we rely on Smokey and Icinga checks to make sure we the site functions.
```

Comment:
I wonder if some of this behaviour should be handled at GDS API Adapters or another appropriate layer? It seems like a bit of a problem if each of the Frontend apps all have the same timeout handling logic?

Reply:
Good point - added in https://github.com/alphagov/gds-api-adapters/issues/776.